drizzlepac/haputils/hapcut_utils.py

```diff
@@ -0,0 +1,324 @@
+
+    # Compute the limits of the cutout region
+    deg_cutout_size = cutout_size.to(u.deg)
+    ra_min = sky_coord.ra.value - deg_cutout_size.value[0]
+    ra_max = sky_coord.ra.value + deg_cutout_size.value[0]
+    dec_min = sky_coord.dec.value - deg_cutout_size.value[1]
+    dec_max = sky_coord.dec.value + deg_cutout_size.value[1]
+
+    # Filter the output as necessary to include only MVM filenames (MVM prefix: hst_skycell).
+    # Also, filter out images which are not actually in the requested cutout region as the
+    # archive search had to be done using a radius.
+    good_rows = []
+    updated_table = None
+    for i, old_row in enumerate(query_table):
+        #if old_row["obs_id"].startswith("hst_skycell"):
+        if old_row["obs_id"].startswith("hst_"):
+            if old_row["s_ra"] >= ra_min and old_row["s_ra"] <= ra_max and \
+               old_row["s_dec"] >= dec_min and old_row["s_dec"] <= dec_max:
+                good_rows.append(old_row)
+    updated_query_table = Table(rows=good_rows, names=query_table.colnames)
+
+    # Catch the case where no files are found which satisfy the criteria
+    if not updated_query_table:
+        print("WARNING: Query for objects within {} of {} returned NO RESULTS!".format(radius, sky_coord))
+        return updated_query_table
+
+    # Get the data product list associated with the elements of the table
+    if verbose:
+        print("Get the product list for all entries in the query table.")
+    dp_table = Observations.get_product_list(updated_query_table)
+
+    # Filter on MVM drizzled product
+    suffix = ["DRZ", "DRC"]
+    if verbose:
+        print("Filter the product list table for only {} filenames.".format(suffix))
+    filtered_dp_table = Observations.filter_products(dp_table,
+                                                     productSubGroupDescription=suffix,
+                                                     extension="fits")
+
+    if not filtered_dp_table:
+        print("WARNING: No drizzle product files (DRZ/DRC) found.")
+        return filtered_dp_table
+
+    # Write the filtered data product table out to a file.  This allows for further
+    # manipulation of the information before a list of filenames is distilled from
+    # the table.
+    # Output filename in the form: mvm_query_ra.dddd_sdec.dddd_radius_cutout.ecsv.
+    #                              mvm_query_84.9208_s69.1483_71_cutout.ecsv
+    ns = "s" if sky_coord.dec.degree < 0.0 else "n"
+    query_filename = "mvm_query_" + str(sky_coord.ra.degree) + "_" + ns + str(abs(sky_coord.dec.degree)) + \
+                     "_{:.0f}_cutout".format(radius.value) + ".ecsv" 
+
+    if verbose:
+        print("Writing out the filter product list table to {}.".format(query_filename))
+    filtered_dp_table.write(query_filename, format="ascii.ecsv")
+
+    return filtered_dp_table
+
+
+def mvm_retrieve_files(products, archive=False, clobber=False, verbose=False):
```

Comment:
The name of this function implies that it will actually get the images themselves (files) as opposed to only reporting the filenames back to the user.  'mvm_retrieve_filenames' would be clearer.

Reply:
Unless the file is already found on disk, this function does retrieve the files.

Reply:
Please update the docstring to make this behavior (more?) apparent.

Reply:
Fixed in 6620833.